src/corebluetooth/central_delegate.rs

```diff
@@ -340,6 +354,51 @@ pub mod CentralDelegate {
             delegate,
             CentralDelegateEvent::DiscoveredPeripheral(held_peripheral),
         );
+
+        let puuid_nsstring = ns::uuid_uuidstring(cb::peer_identifier(peripheral));
+        let puuid = Uuid::from_str(&NSStringUtils::string_to_string(puuid_nsstring)).unwrap();
+
+        let manufacturer_data = ns::dictionary_objectforkey(_adv_data, unsafe {
```

Comment:
Now that you are using `_adv_data`, remote the underscore at the start. It was just there to indicate that the parameter was unused.